library/webpack-plugins/react-universal-plugin.js

```diff
@@ -169,39 +169,24 @@ module.exports = function reactUniversalPlugin () {
             source,
             '',
             '// __webpack_js_chunk_information__',
-            `${this.requireFn}.jci = ${JSON.stringify({ universalChunkNames, manifest })};`
+            `${compilation.mainTemplate.requireFn}.jci = ${JSON.stringify({ universalChunkNames, manifest })};`
           ]
-          return this.asString(buf)
-        })
-        compilation.mainTemplate.plugin('global-hash', () => true)
-        normalModuleFactory.plugin('parser', (parser, parserOptions) => {
-          parser.plugin(`expression __webpack_js_chunk_information__`, ParserHelpers.toConstantDependency('__webpack_require__.jci'))
-          parser.plugin(`evaluate typeof __webpack_js_chunk_information__`, ParserHelpers.evaluateToString('array'))
+          return buf.join('\n')
         })
+        compilation.mainTemplate.hooks.globalHash.tap(p, () => true)
+        normalModuleFactory.hooks.parser.for('javascript/auto').tap(p, addParserHooks)
+        normalModuleFactory.hooks.parser.for('javascript/dynamic').tap(p, addParserHooks)
```

Comment:
I notice there's duplicate code in `merge-css-plugin`

Reply:
What is the duplication?

Reply:
```js
compilation.dependencyFactories.set(ConstDependency, new NullFactory())
compilation.dependencyTemplates.set(ConstDependency, new ConstDependency.Template())
compilation.mainTemplate.hooks.requireExtensions.tap(p, function(source, chunk, hash) {
  // ...
}
compilation.mainTemplate.hooks.globalHash.tap(p, () => true)
normalModuleFactory.hooks.parser.for('javascript/auto').tap(p, addParserHooks)
normalModuleFactory.hooks.parser.for('javascript/dynamic').tap(p, addParserHooks)
```
Might be just fine like this

Reply:
Ahh, yeah, the code looks very similar. It even is similar, but I am unsure if I want to extract it. I'll experiment with it.

Reply:
I changed it, despite the amount of arguments required I think it's a good abstraction. It also showed a few minor flaws so it already paid off (or proved that `evaluateTypeOf` was not used).

Reply:
Haha, check what I found (committed at Oct 1, 2017): https://github.com/kaliberjs/build/blob/aec93e775f305362b15acf864741adb49e6ef31d/library/lib/webpack-utils.js

The code is very similar, haha: https://github.com/kaliberjs/build/blob/89b75cf5ade159f4dfb4cc8ab9ff4ad247a9db1a/library/lib/webpack-utils.js